### PR TITLE
Updates Druid versions for 24.7

### PR DESCRIPTION
**Pull request by lfrancke**
# Description

- Adds 30.0.0, removes 27.0.0
- Fixes run-druid script to include parameters needed for Java 11 and above
- Switch to Java 17 for runtime
- Switches the Dockerfile to heredoc format for easier reading
- Installs "native" pyyaml instead of via pip
- Adds package_manifest
- Removes outdated comment from base image
- Fixes typo in upload script
- Prometheus Emitter is built from source as well while all other community extensions are skipped
- Patch to update all dependencies that have a patch-level update
- Remove log4shell workarounds. Druid 26.0.0 (and later) use Log4j 2.18 (or later) which is not vulnerable anymore
- The dependencies needed for the XmlLayout are now included in the build itself so we don't need to download them from Nexus anymore and we can be sure that the dependencies match up
- Update labels

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
- [x] All added packages (via microdnf or otherwise) have a comment on why they are added
- [x] Things not downloaded from Red Hat repositories should be mirrored in the Stackable repository and downloaded from there
- [x] All packages should have (if available) signatures/hashes verified
- [x] Add an entry to the CHANGELOG.md file
- [ ] Integration tests ran successfully
```

<details>
<summary>TIP: Running integration tests with a new product image</summary>

The image can be built and uploaded to the kind cluster with the following commands:

```shell
bake --product <product> --image-version <stackable-image-version>
kind load docker-image <image-tagged-with-the-major-version> --name=<name-of-your-test-cluster>
```

See the output of `bake` to retrieve the image tag for `<image-tagged-with-the-major-version>`.
</details>

### CODE REVIEW

**Comment by lfrancke** [comment]
For anyone wanting to thes this.
I have pushed a version of the image as of commit https://github.com/stackabletech/docker-images/pull/731/commits/476c71c6435ea6e9efb93140343eb533fa9d5fc6 to

* `docker.stackable.tech/sandbox/druid:26.0.0-stackable0.0.0-dev`
* `docker.stackable.tech/sandbox/druid:28.0.1-stackable0.0.0-dev`
* `docker.stackable.tech/sandbox/druid:30.0.0-stackable0.0.0-dev`

File: druid/stackable/patches/26.0.0/03-stop-building-unused-extensions.patch [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+-        <module>extensions-contrib/materialized-view-selection</module>
+-        <module>extensions-contrib/momentsketch</module>
+-        <module>extensions-contrib/moving-average-query</module>
+-        <module>extensions-contrib/tdigestsketch</module>
+-        <module>extensions-contrib/influxdb-emitter</module>
+-        <module>extensions-contrib/gce-extensions</module>
+-        <module>extensions-contrib/aliyun-oss-extensions</module>
+         <module>extensions-contrib/prometheus-emitter</module>
+-        <module>extensions-contrib/opentelemetry-emitter</module>
+-        <module>extensions-contrib/kubernetes-overlord-extensions</module>
```
[the remark this answers is not in the document]
Comment (lfrancke) [reply druid/stackable/patches/26.0.0/03-stop-building-unused-extensions.patch:47]: There is no change to before here.

So what used to happen is that ALL extensions were built but then at the distribution step only a select few are included in the final build.

What this change does is that it does not even build the things we're not including anyway.

So customers can do the same things as before and before they also could not include these things.

**Comment by lfrancke** [comment]
Sigi and I tested integration tests (I ran various tests and sigi tried the logging tests) and they were all successful

File: stackable-base/Dockerfile [hunk 2]
```diff
@@ -113,3 +110,30 @@ RUN update-ca-trust && \
...
+LABEL io.openshift.expose-services=""
+
+# https://github.com/opencontainers/image-spec/blob/036563a4a268d7c08b51a08f05a02a0fe74c7268/annotations.md#annotations
+LABEL org.opencontainers.image.authors="info@stackable.tech"
+LABEL org.opencontainers.image.url="https://stackable.tech"
+LABEL org.opencontainers.image.source="https://github.com/stackabletech/docker-images/"
+LABEL org.opencontainers.image.vendor="Stackable GmbH"
+LABEL org.opencontainers.image.licenses="Apache-2.0"
+
+COPY --chown=stackable:stackable shared/npm/.npmrc /stackable/
```
Comment (dervoeti) [inline stackable-base/Dockerfile:139]: I don't get this line. Why is it needed and where is it `.npmrc` copied from?

Comment (lfrancke) [reply stackable-base/Dockerfile:139]: Oh, damn. That is a mistake and shoud not be in there. This is a test to try the build-repo. I'll remove it. Good catch. Thank you!

Comment (lfrancke) [reply stackable-base/Dockerfile]: [`4c7dc6e` (#731)](https://github.com/stackabletech/docker-images/pull/731/commits/4c7dc6e9287d15dbf2d45494d718616ba5920843)

File: druid/Dockerfile [hunk 3]
```diff
@@ -2,95 +2,111 @@
...
-    python-pip \
-    # Required to patch druid
-    patch && \
-    microdnf clean all && \
-    rm -rf /var/cache/yum && \
-    # pyyaml is required for the compile Druid
-    pip install --no-cache-dir pyyaml==6.0.1
+# Setting this to anything other than "true" will keep the cache folders around (e.g. for Maven, NPM etc.)
+# This can be used to speed up builds when disk space is of no concern.
+ARG DELETE_CACHES="true"
```
Comment (razvan) [inline druid/Dockerfile:19]: This can be moved to `druid/versions.py` now so it can be overridden with
```
bake --build-arg 'delete_caches=false' ...
```

**Review by razvan — COMMENTED** [review]
Not a full review, just this one remark

File: druid/Dockerfile (hunk 3 above)
Comment (lfrancke) [reply druid/Dockerfile:19]: Oh...it needs to be in `versions.py` so the override works?
That's a bit annoying, I don't want to pollute `versions.py`

Comment (lfrancke) [reply druid/Dockerfile:19]: Razvan and I talked and decided that this is a larger discussion that we don't have to have now.
I'll resolve this for now. It is usable for those that know how to...we can improve it later.

**Review by dervoeti — COMMENTED** [review]
One last very small thing, the rest looks good, I'll approve the PR after that.

**Review by dervoeti — APPROVED** [review]
LGTM

**Pull request merged**